Client/game_sa/CFxSA.cpp

```diff
@@ -237,3 +256,33 @@ void CFxSA::TriggerFootSplash(CVector& vecPosition)
         call    dwFunc
     }
 }
+
+bool CFxSA::IsShadowsLimitReached()
+{
+    short shadows = *(short*)0xC403DC;
+    return shadows >= 48;
+}
```

Comment:
Seems dodgy. I don't know if `game_sa` is allowed to have something like this, or if you should be accessing a field of some class properly.

Reply:
The address points to `CShadows::ShadowsStoredToBeRendered`. The type of the global variable is `unsigned short` instead of `signed short`. This small part of the code can be fixed and refactored, like this:
```
bool CFxSA::IsShadowsLimitReached()
{
    unsigned short& CShadows_ShadowsStoredToBeRendered = *(unsigned short*)0xC403DC;
    return CShadows_ShadowsStoredToBeRendered >= 48;
}
```
Not the best way, but a bit better since this is the actual name. Btw, direct memory access is allowed game_sa because it's similar to multiplayer_sa. I don't blame him for this, we don't any proper coding guidelines about gtasa memory access. I'll create an issue for this tomorrow and explain my idea for handling direct memory access in game_sa and multiplayer_sa.